go/cmd/dolt/commands/schcmds/import.go

```
@@ -16,26 +16,313 @@ package schcmds
+		return errhand.BuildDError("error: missing required parameter.").AddDetails("Must provide exactly one of the operation flags '--create', or '--replace'").SetPrintUsage().Build()
+	} else if apr.Contains(updateFlag) {
+		if apr.ContainsAny(createFlag, replaceFlag) {
+			return errhand.BuildDError("error: multiple operations supplied").AddDetails("Only one of the flags '--create', '--update', or '--replace' may be provided").SetPrintUsage().Build()
+		}
+		op = updateOp
+	} else if apr.Contains(replaceFlag) {
+		if apr.Contains(createFlag) {
+			return errhand.BuildDError("error: multiple operations supplied").AddDetails("Only one of the flags '--create', '--update', or '--replace' may be provided").SetPrintUsage().Build()
+		}
+		op = replaceOp
+	} else {
+		if apr.Contains(keepTypesParam) {
+			return errhand.BuildDError("error: parameter keep-types not supported for create operations").AddDetails("keep-types parameter is used to keep the existing column types as is without modification.").Build()
+		}
+	}
+
+	tbl, tblExists, err := root.GetTable(ctx, tblName)
+
+	if err != nil {
+		return errhand.BuildDError("error: failed to read from database.").AddCause(err).Build()
+	} else if tblExists && op == createOp {
+		return errhand.BuildDError("error: failed to create table.").AddDetails("A table named '%s' already exists.", tblName).AddDetails("Use --replace or --update instead of --create.").Build()
+	}
+
+	var existingSch schema.Schema = schema.EmptySchema
+	if tblExists {
+		existingSch, err = tbl.GetSchema(ctx)
+
+		if err != nil {
+			return errhand.BuildDError("error: failed to read schema from '%s'", tblName).AddCause(err).Build()
+		}
+	}
+
+	val, pksOK := apr.GetValue(pksParam)
+
+	var pks []string
+	if pksOK {
+		pks = strings.Split(val, ",")
+		goodPKS := make([]string, 0, len(pks))
+
+		for _, pk := range pks {
+			pk = strings.TrimSpace(pk)
+
+			if pk != "" {
+				goodPKS = append(goodPKS, pk)
+			}
```

Probably should throw an error on malformed pk strings, no?

---

added more errors